boards/arm/arduino_portenta_h7/arduino_portenta_h7_m7.dts

```diff
@@ -0,0 +1,99 @@
+/*
+ * Copyright (c) 2022 Benjamin Björnsson <benjamin.bjornsson@gmail.com>.
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+/dts-v1/;
+#include <st/h7/stm32h747Xi_m7.dtsi>
+#include <st/h7/stm32h747xihx-pinctrl.dtsi>
+#include "arduino_portenta_h7-common.dtsi"
+
+/ {
+	model = "Arduino Portenta H7 board";
+	compatible = "arduino,portenta-h7";
+
+	/* HW resources are split between CM7 and CM4 */
+	chosen {
+		zephyr,console = &usart1;
+		zephyr,bt-uart = &usart1;
+		zephyr,shell-uart = &usart1;
+		zephyr,sram = &sram0;
+		zephyr,flash = &flash0;
+		zephyr,code-partition = &code_partition;
+	};
+
+	oscen: oscen {
+		compatible = "regulator-fixed-sync", "regulator-fixed";
+		label = "oscen";
+		regulator-name = "oscen";
+		enable-gpios = <&gpioh 1 GPIO_ACTIVE_HIGH>;
+		regulator-boot-on;
+	};
+};
+
+&clk_hse {
+	status = "okay";
+	clock-frequency = <DT_FREQ_M(25)>;
+	hse-bypass;
+};
+
+&pll {
+	div-m = <4>;
+	mul-n = <64>;
+	div-p = <4>;
+	div-q = <2>;
+	div-r = <2>;
+	clocks = <&clk_hse>;
+	status = "okay";
+};
+
+&rcc {
+	clocks = <&pll>;
+	clock-frequency = <DT_FREQ_M(100)>;
```

Comment:
Is there a reason to not set the clock to its maximal value of 480 MHz?

Reply:
I took a look at this again which raised an issue. When trying to run the board at a higher frequency the UART gets corrupted.

My guess is that it has to do with the 25 MHz oscillator and that it's somehow inactive? I added the oscen regulator to the DTS to pull the OSCEN high during boot but is this enough?
I also just noted that in the Arduino source code they write to the PMIC over I2C to activate the 3V1 rail, which is the voltage source for the oscillator. I have tried to write to the same address in another SYS_INIT but the UART is still corrupted at higher frequencies...